xds/internal/resolver/xds_resolver_test.go

```diff
@@ -116,20 +117,45 @@ func (s) TestResolverBuilder(t *testing.T) {
 	tests := []struct {
 		name          string
 		xdsClientFunc func() (xdsclient.XDSClient, error)
+		target        resolver.Target
 		wantErr       bool
 	}{
 		{
 			name: "simple-good",
 			xdsClientFunc: func() (xdsclient.XDSClient, error) {
 				return fakeclient.NewClient(), nil
 			},
+			target:  target,
 			wantErr: false,
 		},
 		{
 			name: "newXDSClient-throws-error",
 			xdsClientFunc: func() (xdsclient.XDSClient, error) {
 				return nil, errors.New("newXDSClient-throws-error")
 			},
+			target:  target,
+			wantErr: true,
+		},
+		{
+			name: "authority not defined in bootstrap",
+			xdsClientFunc: func() (xdsclient.XDSClient, error) {
+				c := fakeclient.NewClient()
+				c.SetBootstrapConfig(&bootstrap.Config{
+					ClientDefaultListenerResourceNameTemplate: "%s",
+					Authorities: map[string]*bootstrap.Authority{
+						"test-authority": {
+							ClientListenerResourceNameTemplate: "xdstp://test-authority/%s",
+						},
+					},
+				})
+				return c, nil
+			},
+			target: resolver.Target{
+				URL: url.URL{
+					Host: "non-existing-authority",
+					Path: "/" + targetStr,
+				},
+			},
```

Comment:
Do you think it would be better to pass a target string to `Dial` instead of using a `resolver.Target` and calling `Build()` directly?

Doing the former would make these tests not depend on the actual parsing logic in `Dial()`.

Reply:
Those are resolver unit tests. I think it makes sense to call Build().

Tests calling Dial() are covered by the e2e tests. (Not for federation yet)